test/unit_tests/braket/circuits/test_box_drawing_circuit_diagram.py

```diff
@@ -0,0 +1,986 @@
+        "      └───┘   │   ",
+        "            ┌─┴─┐ ",
+        "q2 : ───────┤ X ├─",
+        "            └───┘ ",
+        "T  : │  0  │  1  │",
+    )
+    _assert_correct_diagram(circ, expected)
+
+
+def test_single_qubit_result_types_target_none():
+    circ = Circuit().h(0).probability()
+    expected = (
+        "T  : │  0  │ Result Types  │",
+        "      ┌───┐ ┌─────────────┐ ",
+        "q0 : ─┤ H ├─┤ Probability ├─",
+        "      └───┘ └─────────────┘ ",
+        "T  : │  0  │ Result Types  │",
+    )
+    _assert_correct_diagram(circ, expected)
+
+
+def test_result_types_target_none():
+    circ = Circuit().h(0).h(100).probability()
+    expected = (
+        "T    : │  0  │ Result Types  │",
+        "        ┌───┐ ┌─────────────┐ ",
+        "q0   : ─┤ H ├─┤ Probability ├─",
+        "        └───┘ └──────┬──────┘ ",
+        "        ┌───┐ ┌──────┴──────┐ ",
+        "q100 : ─┤ H ├─┤ Probability ├─",
+        "        └───┘ └─────────────┘ ",
+        "T    : │  0  │ Result Types  │",
+    )
+    _assert_correct_diagram(circ, expected)
+
+
+def test_result_types_target_some():
+    circ = (
+        Circuit()
+        .h(0)
+        .h(1)
+        .h(100)
+        .expectation(observable=Observable.Y() @ Observable.Z(), target=[0, 100])
+    )
+    expected = (
+        "T    : │  0  │    Result Types    │",
+        "        ┌───┐ ┌──────────────────┐ ",
+        "q0   : ─┤ H ├─┤ Expectation(Y@Z) ├─",
+        "        └───┘ └────────┬─────────┘ ",
+        "        ┌───┐          │           ",
+        "q1   : ─┤ H ├──────────┼───────────",
+        "        └───┘          │           ",
+        "        ┌───┐ ┌────────┴─────────┐ ",
+        "q100 : ─┤ H ├─┤ Expectation(Y@Z) ├─",
+        "        └───┘ └──────────────────┘ ",
+        "T    : │  0  │    Result Types    │",
+    )
+    _assert_correct_diagram(circ, expected)
+
+
+def test_additional_result_types():
+    circ = Circuit().h(0).h(1).h(100).state_vector().amplitude(["110", "001"])
+    expected = (
+        "T    : │  0  │",
+        "        ┌───┐ ",
+        "q0   : ─┤ H ├─",
+        "        └───┘ ",
+        "        ┌───┐ ",
+        "q1   : ─┤ H ├─",
+        "        └───┘ ",
+        "        ┌───┐ ",
+        "q100 : ─┤ H ├─",
+        "        └───┘ ",
+        "T    : │  0  │",
+        "",
+        "Additional result types: StateVector, Amplitude(110,001)",
+    )
+    _assert_correct_diagram(circ, expected)
+
+
+def test_multiple_result_types():
+    circ = (
+        Circuit()
+        .cnot(0, 2)
+        .cnot(1, 3)
+        .h(0)
+        .variance(observable=Observable.Y(), target=0)
+        .expectation(observable=Observable.Y(), target=2)
+        .sample(observable=Observable.Y())
+    )
+    expected = (
+        "T  : │     0     │  1  │          Result Types          │",
+        "                  ┌───┐   ┌─────────────┐  ┌───────────┐ ",
+        "q0 : ───●─────────┤ H ├───┤ Variance(Y) ├──┤ Sample(Y) ├─",
```

Comment:
Shouldn't the `Y` observables be unlinked? This should be four separate Y measurements

Reply:
For my own context, what does unlinked mean here?

Reply:
Oh got it, in the same way that `Circuit.h([0, 1, 2, 3])` wouldn't put a link between the gates because they are separate 1-qubit operations